src/providers/sh/commands/deploy.js

```diff
@@ -782,6 +791,89 @@ async function readMeta(
   }
 }
 
+async function printEvents(now, token, teamId, onOpen = ()=>{}) {
+  const { _id: id } = now;
+  let url = `https://${apiUrl}/v1/now/deployments/${id}/events?follow=1`;
+  if (teamId) {
+    url += `&teamId=${teamId}`;
+  }
+
+  // we keep track of how much we log in case we
+  // drop the connection and have to start over
+  let o = 0;
+
+  await retry(async bail => {
+    // if we are retrying, we clear past logs
+    if (!quiet && o) process.stdout.write(eraseLines(0));
+
+    const res = await now._fetch(url);
+    if (res.ok) {
+      // fire the open callback and ensure it's only fired once
+      onOpen();
+      onOpen = ()=>{};
+
+      // handle the event stream and make the promise get rejected
+      // if errors occur so we can retry
+      return new Promise((resolve, reject) => {
+        const stream = res.body.pipe(jsonlines.parse());
+        const onData = ({ type, payload }) => {
+          // if we are 'quiet' because we are piping, simply
+          // wait for the first instance to be started
+          // and ignore everything else
+          if (quiet) {
+            if (type === 'instance-start') {
+              resolve();
+            }
+            return;
+          }
+
+          switch (type) {
+            case 'build-start':
```

Comment:
This is waiting on an API improvement by @arunoda

Reply:
@rauchg here's the fix: https://github.com/zeit/now-atlas-events/pull/16 
Also krolled this into BRU.